### PR TITLE
[#170234614] refactor build system and docs

### PR DESCRIPTION
**Pull request by gunzip**
This PR lets you run the frontend application seamlessy with or without docker with hot module reloading enabled in both cases.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,159 +1,136 @@
...
 ```shell
-# Copy the SPID demo configurations folder from the application backend repository
-git clone https://github.com/teamdigitale/io-onboarding-pa-api.git
-cp -r io-onboarding-pa-api/certs .
-cp -r io-onboarding-pa-api/testenv2 .
-rm -rf io-onboarding-pa-api
-
-# Bring up the Docker test environment (build the frontend, pull the backend and the SPID test environment)
-docker-compose up
+# run yarn inside docker container
```
Comment (giux78) [inline README.md]: i had to chmod +x ./scripts/yarn.sh

Comment (giux78) [inline README.md]: instead # Launch docker image and run yarn inside docker 

for me was not trivial to understand without going into the scripts

File: README.md [hunk 2]
```diff
@@ -1,159 +1,136 @@
...
+[https://pa-onboarding.dev.io.italia.it/](https://pa-onboarding.dev.io.italia.it/).
 
 ## Test the application locally
 
-The application can be built and tested locally, either directly on the developer machine (directly using development tools, such as *yarn* or *parcel*), or as a Docker-based image.
-
-### Test locally with mocked APIs
-
-To test the frontend application locally, on developer machine, against some mocked APIs, run in the root folder
+Set up your environment variables first:
```
Comment (giux78) [inline README.md]: with env.example copied everything should work with the mocked server?

File: README.md [hunk 3]
```diff
@@ -1,159 +1,136 @@
...
 
 ```shell
-yarn start-with-mock
+cp env.example .env
+# edit .env file
 ```
```
Comment (giux78) [inline README.md]: with  the copied env.example into .env the mocked server and the app should work without any changes?

Comment (gunzip) [reply README.md]: yes, it should

File: README.md (hunk 1 above)
Comment (gunzip) [reply README.md]: fixed

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This PR changes a total of 634 LOCs, that is more than a reasonable size of 250. Consider splitting the pull request into smaller ones.</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>:tada: - congrats on your new release</td>
    </tr>
  </tbody>
</table>


## Affected stories

  * 🌟 [#170234614](https://www.pivotaltracker.com/story/show/170234614): Come DEV voglio poter sviluppare in locale il frontend utilizzando dei dati fittizi


New dependencies added: <a href='https://www.npmjs.com/package/dotenv-cli'><code>dotenv-cli</code></a>, <a href='https://www.npmjs.com/package/npm-run-all'><code>npm-run-all</code></a>, <a href='https://www.npmjs.com/package/parcel-plugin-html-externals'><code>parcel-plugin-html-externals</code></a> and <a href='https://www.npmjs.com/package/shx'><code>shx</code></a>.


<h2><a href="https://github.com/entropitor/dotenv-cli#readme">dotenv-cli</a></h2>
<p>Author: entropitor</p>
<p>Description: A global executable to run applications with the ENV variables loaded by dotenv</p>
<p>Homepage: <a href="https://github.com/entropitor/dotenv-cli#readme">https://github.com/entropitor/dotenv-cli#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>over 4 years ago</td></tr><tr><td>Last Updated</td><td>21 days ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>10</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/cross-spawn'><code>cross-spawn</code></a>, <a href='https://www.npmjs.com/package/dotenv'><code>dotenv</code></a>, <a href='https://www.npmjs.com/package/dotenv-expand'><code>dotenv-expand</code></a> and <a href='https://www.npmjs.com/package/minimist'><code>minimist</code></a></td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

# dotenv-cli

## Installing

NPM
```bash
$ npm install -g dotenv-cli
```

Yarn
```bash
$ yarn global add dotenv-cli
```

## Usage

```bash
$ dotenv <command with arguments>
```

This will load the variables from the .env file in the current working directory and then run the command (using the new set of environment variables).

### Custom .env files
Another .env file could be specified using the -e flag:
```bash
$ dotenv -e .env2 <command with arguments>
```

Multiple .env files can be specified, and will be processed in order:
```bash
$ dotenv -e .env3 -e .env4 <command with arguments>
```

### Check env variable
If you want to check the value of an environment variable, use the `-p` flag
```bash
$ dotenv -p NODE_ENV
```

### Flags to the underlying command
If you want to pass flags to the inner command use `--` after all the flags to `dotenv-cli`. 

E.g. the following command without dotenv-cli:
```bash
mvn exec:java -Dexec.args="-g -f"
```

will become the following command with dotenv-cli:
```bash
$ dotenv -- mvn exec:java -Dexec.args="-g -f"
``` 
or in case the env file is at `.my-env`
```bash
$ dotenv -e .my-env -- mvn exec:java -Dexec.args="-g -f"
``` 

### Variable expansion
We support expanding env variables inside .env files (See [dotenv-expand](https://github.com/motdotla/dotenv-expand) npm package for more information)

For example:
```
IP=127.0.0.1
PORT=1234
APP_URL=http://${IP}:${PORT}
```
Using the above example `.env` file, `process.env.APP_URL` would be `http://127.0.0.1:1234`.

## License

[MIT](https://en.wikipedia.org/wiki/MIT_License)



</details>




<h2><a href="https://github.com/mysticatea/npm-run-all">npm-run-all</a></h2>
<p>Author: Toru Nagashima</p>
<p>Description: A CLI tool to run multiple npm-scripts in parallel or sequential.</p>
<p>Homepage: <a href="https://github.com/mysticatea/npm-run-all">https://github.com/mysticatea/npm-run-all</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>over 4 years ago</td></tr><tr><td>Last Updated</td><td>13 days ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>58</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/ansi-styles'><code>ansi-styles</code></a>, <a href='https://www.npmjs.com/package/chalk'><code>chalk</code></a>, <a href='https://www.npmjs.com/package/cross-spawn'><code>cross-spawn</code></a>, <a href='https://www.npmjs.com/package/memorystream'><code>memorystream</code></a>, <a href='https://www.npmjs.com/package/minimatch'><code>minimatch</code></a>, <a href='https://www.npmjs.com/package/pidtree'><code>pidtree</code></a>, <a href='https://www.npmjs.com/package/read-pkg'><code>read-pkg</code></a>, <a href='https://www.npmjs.com/package/shell-quote'><code>shell-quote</code></a> and <a href='https://www.npmjs.com/package/string.prototype.padend'><code>string.prototype.padend</code></a></td></tr><tr><td>Keywords</td><td>cli, command, commandline, tool, npm, npm-scripts, run, sequential, serial, parallel and task</td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

| index | [npm-run-all] | [run-s] | [run-p] | [Node API] |
|-------|---------------|---------|---------|------------|

# npm-run-all

[npm version](https://www.npmjs.com/package/npm-run-all)
[Downloads/month](http://www.npmtrends.com/npm-run-all)
[Build Status](https://travis-ci.org/mysticatea/npm-run-all)
[Build status](https://ci.appveyor.com/project/mysticatea/npm-run-all/branch/master)
[Coverage Status](https://codecov.io/gh/mysticatea/npm-run-all)
[Dependency Status](https://david-dm.org/mysticatea/npm-run-all)

A CLI tool to run multiple npm-scripts in parallel or sequential.

## ⤴️ Motivation

- **Simplify.** The official `npm run-script` command cannot run multiple scripts, so if we want to run multiple scripts, it's redundant a bit. Let's shorten it by glob-like patterns.<br>
  Before: `npm run clean && npm run build:css && npm run build:js && npm run build:html`<br>
  After: `npm-run-all clean build:*`
- **Cross platform.** We sometimes use `&` to run multiple command in parallel, but `cmd.exe` (`npm run-script` uses it by default) does not support the `&`. Half of Node.js users are using it on Windows, so the use of `&` might block contributions. `npm-run-all --parallel` works well on Windows as well.

## 💿 Installation

```bash
$ npm install npm-run-all --save-dev
# or
$ yarn add npm-run-all --dev
```

- It requires `Node@>=4`.

## 📖 Usage

### CLI Commands

This `npm-run-all` package provides 3 CLI commands.

- [npm-run-all]
- [run-s]
- [run-p]

The main command is [npm-run-all].
We can make complex plans with [npm-run-all] command.

Both [run-s] and [run-p] are shorthand commands.
[run-s] is for sequential, [run-p] is for parallel.
We can make simple plans with those commands.

#### Yarn Compatibility

If a script is invoked with Yarn, `npm-run-all` will correctly use Yarn to execute the plan's child scripts.

### Node API

This `npm-run-all` package provides Node API.

- [Node API]

## 📰 Changelog

- https://github.com/mysticatea/npm-run-all/releases

## 🍻 Contributing

Welcome♡

### Bug Reports or Feature Requests

Please use GitHub Issues.

### Correct Documents

Please use GitHub Pull Requests.

I'm not familiar with English, so I especially thank you for documents' corrections.

### Implementing

Please use GitHub Pull Requests.

There are some npm-scripts to help developments.

- **npm test** - Run tests and collect coverage.
- **npm run clean** - Delete temporary files.
- **npm run lint** - Run ESLint.
- **npm run watch** - Run tests (not collect coverage) on every file change.

[npm-run-all]: docs/npm-run-all.md
[run-s]: docs/run-s.md
[run-p]: docs/run-p.md
[Node API]: docs/node-api.md


</details>




<h2><a href="https://github.com/stoically/parcel-plugin-html-externals#readme">parcel-plugin-html-externals</a></h2>
<p>Author: stoically</p>
<p>Description: ParcelJS plugin to ignore certain script and stylesheet link tags in html files marked as externals</p>
<p>Homepage: <a href="https://github.com/stoically/parcel-plugin-html-externals#readme">https://github.com/stoically/parcel-plugin-html-externals#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>5 months ago</td></tr><tr><td>Last Updated</td><td>5 months ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>1</td></tr><tr><td>Releases</td><td>3</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/minimatch'><code>minimatch</code></a></td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

ParcelJS plugin to ignore certain script and stylesheet link tags in html files marked as externals.

> Parcel will probably support externals out of the box in the future, [related issue here](https://github.com/parcel-bundler/parcel/issues/144).

### Install

```
npm install --save-dev parcel-plugin-html-externals
```

### Usage

Put an `externals` key into your `package.json`, e.g.

```json
{
  "externals": {
    "vendor/**/*": false
  }
}
```

will ignore all script / stylesheet link tags whose src / href attribute starts with `vendor/`, e.g

```html
<script src="vendor/thirdparty.min.js"></script>
```

The externals key can be a string or any glob accepted by [minimatch](https://github.com/isaacs/minimatch),
the value must be `false` to trigger ignoring.


### Caveats

The plugin monkey patches Parcel internals by extending the HTMLAsset class and proxying its ast walker, so it might just break in an upcoming version.


### License

MIT

</details>




<h2><a href="https://github.com/shelljs/shx#readme">shx</a></h2>
<p>Author: Unknown</p>
<p>Description: Portable Shell Commands for Node</p>
<p>Homepage: <a href="https://github.com/shelljs/shx#readme">https://github.com/shelljs/shx#readme</a></p>

<table>
  <thead><tr><th></th><th width="100%"></th></tr></thead>
  <tr><td>Created</td><td>over 3 years ago</td></tr><tr><td>Last Updated</td><td>over 1 year ago</td></tr><tr><td>License</td><td>MIT</td></tr><tr><td>Maintainers</td><td>2</td></tr><tr><td>Releases</td><td>11</td></tr><tr><td>Direct Dependencies</td><td><a href='https://www.npmjs.com/package/es6-object-assign'><code>es6-object-assign</code></a>, <a href='https://www.npmjs.com/package/minimist'><code>minimist</code></a> and <a href='https://www.npmjs.com/package/shelljs'><code>shelljs</code></a></td></tr><tr><td>Keywords</td><td>shelljs, shell, unix, bash, sh, exec, cli and zsh</td></tr>
</table>

<details>
<summary><code>README</code></summary></br>

# Shx

[Travis](https://travis-ci.org/shelljs/shx)
[AppVeyor](https://ci.appveyor.com/project/shelljs/shx/branch/master)
[Codecov](https://codecov.io/gh/shelljs/shx)
[npm version](https://www.npmjs.com/package/shx)
[npm downloads](https://www.npmjs.com/package/shx)

`shx` is a wrapper around [ShellJS](https://github.com/shelljs/shelljs) Unix
commands, providing an easy solution for simple Unix-like, cross-platform
commands in npm package scripts.

## Difference Between ShellJS and shx

- **ShellJS:** Good for writing long scripts, all in JS, running via NodeJS (e.g. `node myScript.js`).
- **shx:** Good for writing one-off commands in npm package scripts (e.g. `"clean": "shx rm -rf out/"`).

## Install

```shell
npm install shx --save-dev
```
This will allow using `shx` in your `package.json` scripts.

## Usage

### Command Line

If you'd like to use `shx` on the command line, install it globally with the `-g` flag.
The following code can be run *either a Unix or Windows* command line:

```Bash
$ shx pwd                       # ShellJS commands are supported automatically
/home/username/path/to/dir

$ shx ls                        # files are outputted one per line
file.txt
file2.txt

$ shx rm *.txt                  # a cross-platform way to delete files!

$ shx ls

$ shx echo "Hi there!"
Hi there!

$ shx touch helloworld.txt

$ shx cp helloworld.txt foobar.txt

$ shx mkdir sub

$ shx ls
foobar.txt
helloworld.txt
sub

$ shx rm -r sub                 # options work as well

$ shx --silent ls fakeFileName  # silence error output
```

All commands internally call the ShellJS corresponding function, guaranteeing
cross-platform compatibility.

### package.json

ShellJS is good for writing long scripts. If you want to write bash-like,
platform-independent scripts, we recommend you go with that.

However, `shx` is ideal for one-liners inside `package.json`:

```javascript
{
  "scripts": {
    "clean": "shx rm -rf build dist && shx echo Done"
  }
}
```

## Unsupported Commands

Due to the differences in execution environments between ShellJS and `shx` (JS vs CLI) some commands are not supported:

| Unsupported command | Recommend workaround |
| ------------------- | -------------------- |
| `shx cd`            | Just use plain old `cd` (it's the same on windows too) |
| `shx pushd`         | Just use plain old `pushd`. Use forward slashes and double-quote the path. (e.g. `pushd "../docs"`. This would fail on Windows without the quotes) |
| `shx popd`          | Just use plain old `popd` |
| `shx dirs`          | No workaround |
| `shx set`           | See below |
| `shx exit`          | Just use plain old `exit` |
| `shx exec`          | Instead of `shx exec cmd`, just use plain old `cmd` |
| `shx ShellString`   | No workaround (but why would you want this?) |

### Shx options

Shx allows you to modify its behavior by passing arguments. Here's a list of
supported options:

| [`set`](https://github.com/shelljs/shelljs#setoptions) flag | [`shell.config`](https://github.com/shelljs/shelljs#configuration) setting | shx command | Effect |
|:---:| --- | --- | --- |
| `-e` | `config.fatal = true` | Not supported | Exit upon first error |
| `-v` | `config.verbose = true` | `shx --verbose cd foo` | Log the command as it's run |
| `-f` | `config.noglob = true` | `shx --noglob cat '*.txt'` | Don't expand wildcards |
| N/A | `config.silent = true` | `shx --silent cd noexist` | Don't show error output |

## Team

| [Nate Fischer](https://github.com/nfischer) | [Ari Porad](http://github.com/ariporad) | [Levi Thomason](https://github.com/levithomason) |
|:---:|:---:|:---:|
| [Nate Fischer](https://github.com/nfischer) | [Ari Porad](http://github.com/ariporad) | [Levi Thomason](https://github.com/levithomason) |


</details>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 979f3f50d475af47dedb603a1e805b74567db521
</p>

**Comment by gunzip** [comment]
> The docker part seems not to work properly. At least following the steps in the README. Seems that or the port are not exposed from the containers but the host from localhost seems unreachable. The browser is not able to find the provided url and keep trying.
> 

this sounds odd since the ports are exposed inside docker-compose.yml. can you please do some debugging ? (which OS are you running as host ?)

> Instead the part without docker works.

**Comment by giux78** [comment]
Sorry, on my tired mac os x has taken some minutes before completely start with docker. In the end works also the docker compose. Great!!!!!

File: env.example [hunk 4]
```diff
@@ -5,3 +5,6 @@ IO_ONBOARDING_PA_API_HOST=http://localhost
 IO_ONBOARDING_PA_API_PORT=3000
 IO_ONBOARDING_PA_SHOW_FAKE_IDP=1
 IO_ONBOARDING_PA_IS_MOCK_ENV=0
```
Comment (d-al-ibm) [inline env.example]: This should be 1 in mocked backend

**Pull request merged**